parachain/runtimes/gargantua/src/ismp.rs

```diff
@@ -103,6 +105,23 @@ impl Get<Option<StateMachine>> for Coprocessor {
 	}
 }
 
+pub struct ParachainStateMachineProvider;
+
+impl ismp_parachain::ParachainStateMachineProvider<Runtime> for ParachainStateMachineProvider {
+	fn state_machine(id: StateMachine) -> Result<Box<dyn StateMachineClient>, Error> {
+		let para_id = match id {
```

Comment:
Leave these preliminary checks back in the Parachain Consensus client implementation  as mentioned above